util/pace/util/quantity.py

```diff
@@ -387,44 +370,16 @@ def sel(self, **kwargs: Union[slice, int]) -> np.ndarray:
         """
         return self.view[tuple(kwargs.get(dim, slice(None, None)) for dim in self.dims)]
 
-    @property
-    def storage(self):
-        """A gt4py storage representing the data in this Quantity.
-
-        Will raise TypeError if the gt4py backend was not specified when initializing
-        this object, either by providing a Storage for data or explicitly specifying
-        a backend.
-        """
-        if self._storage is None:
-            raise TypeError(
-                "gt4py backend was not specified when initializing this object"
-            )
-        return self._storage
-
-    def _initialize_storage(self, data, origin, gt4py_backend: str, mask: Tuple):
-        storage = gt4py.storage.storage.empty(
-            gt4py_backend,
-            default_origin=origin,
-            shape=data.shape,
-            dtype=data.dtype,
-            mask=mask,
-            managed_memory=True,  # required to get GPUStorage with only gpu data copy
+    def _initialize_data(self, data, origin, gt4py_backend: str, dimensions: Tuple):
+        """Allocates an ndarray with optimal memory layout, and copies the data over."""
```

Comment:
This is fine for now. It would be nice if this can avoid allocating new memory if the ndarray is already in the optimal memory layout, or to be able to avoid copy while initializing a Quantity with a non-optimal memory layout (e.g. for NOAA's case).

Reply:
I made a change that checks to see if the data is in the "optimal" layout and skips this if it is. Nice catch!